qrest/conf.py

```diff
@@ -77,22 +84,51 @@ def _validate(self):
             raise RestClientConfigurationError('parameter "multiple" must be boolean')
         if not isinstance(self.description, str):
             raise RestClientConfigurationError('parameter "description" must be string')
+
         if self.exclusion_group:
             if not isinstance(self.exclusion_group, str):
                 raise RestClientConfigurationError("group name must be a string")
             if not self.exclusion_group.strip():
                 raise RestClientConfigurationError("group name must be a string")
+
         if self.default and self.required:
             raise RestClientConfigurationError(
                 "you cannot combine required=True and a default setting"
             )
-        if self.choices and not isinstance(self.choices, list):
-            raise RestClientConfigurationError("choices must be a list")
-        if self.default and self.choices:
-            if self.default not in self.choices:
-                raise RestClientConfigurationError(
-                    "if there is a choices list, default must be in this list"
-                )
+
+        if self.choices:
+            if not isinstance(self.choices, list):
+                raise RestClientConfigurationError("choices must be a list")
+            if self.default:
+                if self.default not in self.choices:
+                    raise RestClientConfigurationError(
+                        "if there is a choices list, default must be in this list"
+                    )
+            if self.example:
+                if self.example not in self.choices:
+                    raise RestClientConfigurationError("example should be one of the choices")
+
+        if self.schema:
+            if not isinstance(self.schema, dict):
```

Comment:
Apparently if `self.schema` exists, it should be a dict. This means its type is `Optional[Dict[Any, Any]]` and not `Optional[str]`.

Reply:
I'll create a separate ticket to fix all type hints. Currently we use the type hints for our own understanding but I'm afraid that if we start checking them, we will find that we don't always use them correctly.